File: backend/app/schemas/tax_rate.py
import uuid
from datetime import date
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class TaxRateCreate(BaseModel):
    name: str
    rate_percent: Decimal
    effective_from: date


class TaxRateRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    rate_percent: Decimal
    effective_from: date
    effective_to: date | None
    is_active: bool

File: frontend/src/types/taxRate.ts
export interface TaxRateRead {
  id: string;
  name: string;
  rate_percent: string;
  effective_from: string;
  effective_to: string | null;
  is_active: boolean;
}

export interface TaxRateCreate {
  name: string;
  rate_percent: string;
  effective_from: string;
}
